src/Microsoft.Data/StringColumn.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A column to hold strings
+    /// </summary>
+    public partial class StringColumn : BaseColumn
+    {
+        private List<List<string>> _stringBuffers = new List<List<string>>(); // To store more than intMax number of strings
+
+        public StringColumn(string name, long length) : base(name, length, typeof(string))
+        {
+            _stringBuffers.Add(new List<string>());
+        }
+
+        public StringColumn(string name, IEnumerable<string> values) : base(name, 0, typeof(string)) {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            if (_stringBuffers.Count == 0)
+            {
+                _stringBuffers.Add(new List<string>());
+            }
+            foreach (var value in values)
+            {
+                Append(value);
+            }
+        }
+
+        public void Append(string value)
+        {
+            var lastBuffer = _stringBuffers[_stringBuffers.Count - 1];
+            if (lastBuffer.Count == int.MaxValue)
+            {
+                lastBuffer = new List<string>();
+                _stringBuffers.Add(lastBuffer);
+            }
+            lastBuffer.Add(value);
+            Length++;
+        }
+        private int GetBufferIndexContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentOutOfRangeException(strings.ColumnIndexOutOfRange, nameof(rowIndex));
+            }
+            int curArrayIndex = 0;
+            int numBuffers = _stringBuffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > _stringBuffers[curArrayIndex].Count)
+            {
+                rowIndex -= _stringBuffers[curArrayIndex].Count;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        public override object this[long rowIndex]
```

Comment:
Why don't we have a `string` typed indexer like we do for `PrimitiveColumn<T>`?

Reply:
Fixed